src/components/Datatable/ControlModule/ControlModule.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { getColor } from '../../../utils/helpers';
+import { FlexContainer } from '../../FlexContainer';
+import { SearchBar } from '../../forms/SearchBar';
+import { ToolsTabItem } from '../ToolsTabItem';
+import { SSCIconNames } from '../../../theme/icons/icons.enums';
+import { ControlModuleProps } from './ControlModule.types';
+import { Filters } from '../../Filters';
+import { renderSuggestionFilter } from '../../forms/SearchBar/SearchSuggestionFormats';
+
+const FiltersContainer = styled(FlexContainer)`
+  background-color: ${getColor('graphite5H')};
+`;
+
+const ControlModule: React.FC<ControlModuleProps> = ({
```

Comment:
in order to fix the TS error in here, you have to change the ControlModule signature in similar fashion like https://github.com/securityscorecard/design-system/blob/alpha/src/components/Datatable/Datatable.tsx#L47-L57

So it should be:
```js
const ControlModule = <D extends Record<string, unknown>>({
  hasSearch = true,
  searchConfig,
  hasFiltering = true,
  filtersConfig,
  defaultIsFilteringOpen = true,
  hasColumnVisibility = true,
  // defaultHiddenColumns = [],
  hasColumnOrdering = true,
  // defaultColumnOrder,
  hasCustomViews = true,
  // defaultCustomView,
  columnVisibilityActions,
  filteringActions,
  columnOrderingActions,
  customViewsActions,
}: ControlModuleProps <D>): React.ReactElement => {
```
And then in `Datatable.tsx` you will use it like
```js
{controlsConfig.isControlsEnabled && (
         <ControlModule<D>
            {...defaultControlsConfig}
          />
        )}         
```

Reply:
Thanks! I used this type and it fixed most of the conflicts. I still have one type error left in ControlModule storybook though which I couldn't remove

Reply:
I was looking into that and it seems there is some kind of issue with validation between prop-types and TypeScript. 
Quick hacky fix, for now, is to remove the `as Meta` in default export in storybook file.